bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -347,14 +356,34 @@ private void doMutation(final Mutation mutation) throws RetriesExhaustedWithDeta
       // requirements are achieved.
       sequenceId = sizeManager.registerOperationWithHeapSize(mutation.heapSize());
     } catch (InterruptedException e) {
+      synchronized (globalExceptions) {
+        // Add the exception to the list of global exceptions and immediately throw a
+        // RetriesExhaustedWithDetailsException.
+        addGlobalException(mutation, e);
+        handleExceptions();
```

Comment:
This now handles the case of concurrent writers, but couldn't a client still specify an error handler other than the default 'throw whenever floats your boat and on a random thread'? I'm not a fan of that error handler so I imagine I would be one of those clients :)